### PR TITLE
Docker observer: skeleton implementation

### PR DESCRIPTION
**Pull request by mstumpfx**
**Description:** This is the initial skeleton implementation for the docker observer.

**Link to tracking Issue:** #4446 

**Testing:** Basic config and factory test

**Documentation:** Added initial README.md

### CODE REVIEW

File: extension/observer/dockerobserver/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Docker Observer Extension
+
+**Status: beta**
+
+The `docker_observer` uses a Docker client and queries the Docker Engine API for running containers.
+
+Requires Docker API Version 1.22+.
+
+The collector will need permissions to access the Docker Engine API.
```
Comment (rmfitzpatrick) [inline extension/observer/dockerobserver/README.md:9]: Is there a docker doc that describes those required?

File: extension/observer/dockerobserver/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+#### `excluded_images`
+
+A list of filters whose matching images are to be excluded. Supports literals, globs, and regex.
+
+default: `[]`
+
+#### `use_hostname_if_present`
+
+If true, the "Config.Hostname" field (if present) of the docker
```
Comment (rmfitzpatrick) [inline extension/observer/dockerobserver/README.md:34]: Could be helpful to mention where `Config.Hostname` and `NetworkSettings.IPAddress` are being obtained from (https://docs.docker.com/engine/api/v1.41/#operation/ContainerInspect or similar?)

**Review by rmfitzpatrick — APPROVED** [review]

File: extension/observer/dockerobserver/README.md (hunk 1 above)
Comment (mstumpfx) [reply extension/observer/dockerobserver/README.md:9]: I couldn't find a good doc but I can add a note that this means read access on the docker socket

File: extension/observer/dockerobserver/README.md (hunk 2 above)
Comment (mstumpfx) [reply extension/observer/dockerobserver/README.md:34]: Agreed - I'll add this in.

**Review by jrcamp — APPROVED** [review]

File: extension/observer/dockerobserver/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# Docker Observer Extension
+
+**Status: beta**
```
Comment (emaderer) [inline extension/observer/dockerobserver/README.md:3]: The status should probably indicate that it's just a skeleton and cannot be used right now.

**Pull request merged**